### DISCUSSION TITLE
Changing Theme fails on Vue 3 PrimeVue

### DISCUSSION THREAD

**Discussion by Aleroms** [primefaces/community#1022] [PrimeVue]
Hello,
I've tried to the best of my abilities to follow the [documentation](https://primevue.org/theming/#switchthemes) on switching themes but have not had success. When I inspect the page, I view changes in the head, specifically the link element, where it does reflect changes. My `href` attribute changes from `<link id="theme-link" rel="stylesheet" href="/themes/lara-light-green/theme.css">` to `<link id="theme-link" rel="stylesheet" href="/themes/lara-dark-green/theme.css">`, however, the actual theme does not change. 

I have created a test app with just this implemented in my repo [changeTheme_PrimeVue](https://github.com/Aleroms/changeTheme_PrimeVue) as well as a [live site](https://changetheme-primevue.netlify.app/) example.

Here are the steps I took to create this below.

how to replicate:
create vue app `npm init vue@latest`
install primevue `npm install primevue`
 in `main.js` 
register plugin and import theme  
`import 'primevue/resources/themes/lara-light-green/theme.css'`
`import "primevue/resources/themes/lara-dark-green/theme.css";`
copy themes from PrimeVue `resources/themes` folder into `public/themes`
place `link` in `index.html` `<link id="theme-link" rel="stylesheet" href="/themes/lara-light-green/theme.css"/>`
copy and paste code from [App.vue](https://github.com/Aleroms/changeTheme_PrimeVue/blob/main/src/App.vue) into `App.vue`
run `dev` `npm run dev`

**Comment by Sun3** [comment]
ps: I am also having the same issue, did you find a resolution?

Thanks

**Comment by Sun3** [comment]
This is the code I am using.

main.ts
```
import 'primevue/resources/themes/aura-dark-cyan/theme.css';
import PrimeVue from 'primevue/config';
app.use(PrimeVue);
```

app.vue
```
import { usePrimeVue } from 'primevue/config';
import { onMounted } from 'vue';

const PrimeVue = usePrimeVue();

onMounted(() => {
  PrimeVue.changeTheme('aura-dark-cyan', 'aura-light-cyan', 'theme-link', () => {});
  // PrimeVue.changeTheme('md-dark-indigo', 'md-light-indigo', 'theme-link', () => {});
})
```

I also tried but did not work:
```
window.matchMedia('(prefers-color-scheme: dark)').addEventListener('change', event => {
  PrimeVue.changeTheme('aura-dark-cyan', 'aura-light-cyan', 'theme-link', () => {});
});
```

**Comment by cagataycivici** [comment]
Themes should be moved to public folder and you should not import it as a module, please make sure this is done. Having said that, PrimeVue v4 theming is superior so we don't need to worry about these.

Documentation covers this in detail.

v3
https://primevue.org/theming/#switchthemes

v4
https://v4.primevue.org/theming/styled

**Comment by Sun3** [comment]
I am using Primevue beta.2 and I love the theme changes automatically according to system...

I am using Vue 3 with Typescript and I receive the following errors. I can still run the app but I was not sure if this is the appropriate place to report issues or somewhere else

Code in main.ts
```
import Aura from 'primevue/themes/aura';

app.use(PrimeVueStyled, {
  // Default theme configuration
  theme: {
      preset: Aura,
      options: {
          prefix: 'p',
          darkModeSelector: 'system',
          cssLayer: false
      }
  }
});
```


Errors:
```
src/main.ts:12:18 - error TS7016: Could not find a declaration file for module 'primevue/themes/aura'. '/Volumes/Data/Pixolini/Vue/Vue Apps/ministry-calendar/node_modules/primevue/themes/aura/index.cjs.js' implicitly has an 'any' type.

12 import Aura from 'primevue/themes/aura';
                    ~~~~~~~~~~~~~~~~~~~~~~

src/main.ts:42:9 - error TS2769: No overload matches this call.
  Overload 1 of 2, '(plugin: Plugin<[{ theme: { preset: any; options: { prefix: string; darkModeSelector: string; cssLayer: boolean; }; }; }]>, options_0: { theme: { preset: any; options: { prefix: string; darkModeSelector: string; cssLayer: boolean; }; }; }): App<...>', gave the following error.
    Argument of type 'typeof import("/Volumes/Data/Pixolini/Vue/Vue Apps/ministry-calendar/node_modules/primevue/styled/PrimeVueStyled")' is not assignable to parameter of type 'Plugin<[{ theme: { preset: any; options: { prefix: string; darkModeSelector: string; cssLayer: boolean; }; }; }]>'.
  Overload 2 of 2, '(plugin: Plugin<{ theme: { preset: any; options: { prefix: string; darkModeSelector: string; cssLayer: boolean; }; }; }>, options: { theme: { preset: any; options: { prefix: string; darkModeSelector: string; cssLayer: boolean; }; }; }): App<...>', gave the following error.
    Argument of type 'typeof import("/Volumes/Data/Pixolini/Vue/Vue Apps/ministry-calendar/node_modules/primevue/styled/PrimeVueStyled")' is not assignable to parameter of type 'Plugin<{ theme: { preset: any; options: { prefix: string; darkModeSelector: string; cssLayer: boolean; }; }; }>'.

42 app.use(PrimeVueStyled, {
           ~~~~~~~~~~~~~~

Found 2 errors.

ERROR: "type-check" exited with 2.
```

Thanks.
